clients/cobol-lsp-vscode-extension/src/__tests__/CopybookDownloadServiceTest.ts

```diff
@@ -12,119 +12,207 @@
+            await (cbd as any).handleQueue(copybookProfile, new Set(), null);
+            expect(vscode.window.showErrorMessage)
+                .toHaveBeenCalledWith("Connection to mainframe using Zowe profile zoweProfile failed.");
+        });
+        it("handleQueue handle No password ZoweError", async () => {
+            const pathGenerator = new CopybooksPathGenerator(null);
+            pathGenerator.listDatasets = jest.fn().mockResolvedValue(["dataset"]);
+            const cbd = new CopybookDownloadService(null, null, null, pathGenerator);
+            (cbd as any).handleDataset = jest.fn().mockRejectedValue(new ZoweError("", Type.NoPassword));
+            vscode.window.showErrorMessage = jest.fn();
+            await (cbd as any).handleQueue(copybookProfile, new Set(), null);
+            expect(vscode.window.showErrorMessage).toHaveBeenCalledWith("No password in Zowe profile zoweProfile.");
+        });
     });
-    it("handleQueue handle Connection refused ZoweError", async () => {
-        const pathGenerator = new CopybooksPathGenerator(null);
-        pathGenerator.listDatasets = jest.fn().mockResolvedValue(["dataset"]);
-        const cbd = new CopybookDownloadService(null, null, null, pathGenerator);
-        (cbd as any).handleDataset = jest.fn().mockRejectedValue(new ZoweError("", Type.ConnRefused));
-        vscode.window.showErrorMessage = jest.fn();
-        await (cbd as any).handleQueue(copybookProfile, new Set(), null);
-        expect(vscode.window.showErrorMessage)
-            .toHaveBeenCalledWith("Connection to mainframe using Zowe profile zoweProfile failed.");
+});
+
+describe("Test the creation of folders that contains copybooks downloaded from MF against correct configuration in settings provided by the user", () => {
+    function setupScenario() {
+        const testFolder = path.join(__dirname, C4Z_FOLDER, COPYBOOKS_FOLDER, "profile", "dataset");
+        const copybookURIPath = path.join(testFolder, "copybook" + ".cpy");
+        const zoweApi = new ZoweApi();
+        const copybooksDownloadService: CopybookDownloadService = new CopybookDownloadService(undefined, zoweApi, undefined, undefined);
+
+        zoweApi.fetchMember = jest.fn().mockReturnValue("");
+        (createCopybookPath as any) = jest.fn().mockReturnValue(copybookURIPath);
+        (createDatasetPath as any) = jest.fn().mockReturnValue(testFolder);
+        return {copybookURIPath, copybooksDownloadService};
+    }
+
+    function cleanupScenario() {
+        fs.remove(path.join(__dirname, C4Z_FOLDER));
+    }
+
+    test("With a valid configuration of dataset, copybook is created on FS", async () => {
+        const {copybookURIPath, copybooksDownloadService} = setupScenario();
+        await (copybooksDownloadService as any).downloadCopybookFromMFUsingZowe("dataset", "copybook", "profile");
+        expect(fs.existsSync(copybookURIPath)).toBe(true);
+
+        // remove folder and file created to run the scenario
```

Comment:
we can remove these comments, method has good naming

Reply:
sure!